### PR TITLE
added hw-31.1

### PR DESCRIPTION
**Pull request by viaviaj**
jdbc, ProductDaoJdbcImpl, ConnectionUtil, init_db.sql

### CODE REVIEW

**Review by julia-shvabovich — APPROVED** [review]
Well done!

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    public Optional<Product> getById(Long id) {
+        String query = "SELECT name, price FROM products WHERE id = ? AND deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String name = resultSet.getString("name");
+                double price = resultSet.getDouble("price");
+                return Optional.of(new Product(name, price));
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: this part of the code should be moved to the separate private method

File: src/main/resources/init_db.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+CREATE DATA internet_shop
+CREATE TABLE `products` (
+  `id` bigint NOT NULL AUTO_INCREMENT,
+  `name` varchar(225) NOT NULL,
+  `price` double NOT NULL,
+  `deleted` tinyint NOT NULL DEFAULT '0',
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `name_UNIQUE` (`name`) )
+  ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=utf8;
```
Comment (Sofasmile) [inline src/main/resources/init_db.sql]: why do use AUTO_INCREMENT=4?

File: src/main/resources/init_db.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+CREATE DATA internet_shop
+CREATE TABLE `products` (
+  `id` bigint NOT NULL AUTO_INCREMENT,
+  `name` varchar(225) NOT NULL,
+  `price` double NOT NULL,
+  `deleted` tinyint NOT NULL DEFAULT '0',
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `name_UNIQUE` (`name`) )
```
Comment (Sofasmile) [inline src/main/resources/init_db.sql]: should product_name be UNIQUE?

**Review by Sofasmile — COMMENTED** [review]
how about main() method?
do you remove dao annotation above the ProductDaoImpl class?

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "UPDATE products SET deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return true;
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: should we always return true?

File: src/main/resources/init_db.sql (hunk 3 above)
Comment (viaviaj) [reply src/main/resources/init_db.sql]: I thought there was only one unique product for each position, for example, Gold_apple
Green_apple

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 5]
```diff
@@ -104,4 +102,11 @@ public boolean deleteById(Long id) {
             throw new DataProcessingException("Failed, list products not found", e);
         }
     }
+
+    private Product getProductFromResSet(ResultSet resultSet) throws SQLException {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java:104]: can we use this method with getAll()?

Comment (viaviaj) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java:104]: agree

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**